path.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logicalcluster
+
+import (
+	"encoding/json"
+	"path"
+	"regexp"
+	"strings"
+)
+
+var (
+	// Wildcard is the Path indicating a requests that spans many logical clusters.
+	Wildcard = Path{value: "*"}
+)
+
+const (
+	separator = ":"
+)
+
+// Path represents a colon separated list of words describing a path in a logical cluster hierarchy,
+// like a file path in a file-system.
+//
+// For example in the case the logical cluster hierarchy is implemented via workspaces,
+// the first word is a logical cluster name, and the following words are workspace names.
+// For instance, in the following hierarchy:
+//
+// root/                    (62208dab)
+// ├── accounting           (c8a942c5)
+// │   └── us-west          (33bab531)
+// │       └── invoices     (f5865fce)
+// └── management           (e7e08986)
+//
+//	└── us-west-invoices    (f5865fce)
+//
+// the following would all be valid paths for the `project` logical cluster:
+//   - root:accounting:us-west:invoices
+//   - 62208dab:accounting:us-west:invoices
+//   - c8a942c5:us-west:invoices
+//   - 33bab531:invoices
+//   - f5865fce
+//   - root:management:us-west-invoices
+//   - 62208dab:management:us-west-invoices
+//   - e7e08986:us-west-invoices
+type Path struct {
```

Comment:
Why do we want this to be a struct? I thought @sttts preferred `type Path string` for fewer allocs?

Reply:
i'll try with `type Path string` I think it should work.
unless you remember why the old type was a struct ?

Reply:
We cannot change this here in the PR. Please leave the struct for now.